testplan/common/utils/comparison.py

```diff
@@ -507,7 +507,10 @@ def should_ignore_key(key: Hashable) -> bool:
         if key in ignore:
             should_ignore = True
         elif only is not None:
-            should_ignore = key not in only
+            if isinstance(only, Iterable):
```

Comment:
I don't think we should override this existing param, that makes the code full of
 if isinstance(only, Iterable)